File: Show_Hide_Password_ObjC/Show_Hide_Password_ObjC/ViewController.m
//
//  ViewController.m
//  Show_Hide_Password_ObjC
//
//  Created by mallikarjun on 09/09/19.
//  Copyright © 2019 Mallikarjun H. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    UIButton *hideShow = [[UIButton alloc] initWithFrame:CGRectMake(0,0,20,20)];
    
    [hideShow setImage:[UIImage imageNamed:@"show"]  forState:UIControlStateNormal];
    
    self.passwordTexyField.secureTextEntry = YES;
    self.passwordTexyField.rightView = hideShow;
    self.passwordTexyField.rightViewMode = UITextFieldViewModeAlways;
    [hideShow addTarget:self action:@selector(hideShow:) forControlEvents:UIControlEventTouchUpInside];

}

- (void)hideShow:(id)sender
{
    UIButton *hideShow = (UIButton *)self.passwordTexyField.rightView;
    
    if (!self.passwordTexyField.secureTextEntry)
    {
        self.passwordTexyField.secureTextEntry = YES;
        [hideShow setImage:[UIImage imageNamed:@"show"] forState:UIControlStateNormal];
    }
    else
    {
        self.passwordTexyField.secureTextEntry = NO;
        [hideShow setImage:[UIImage imageNamed:@"hide"] forState:UIControlStateNormal];
    }
    [self.passwordTexyField becomeFirstResponder];
}

- (IBAction)loginButtonClicked:(id)sender {
    
    NSLog(@"Cliked");
    // write logic for your login method
}
@end

File: Show_Hide_Password_Swift/Show_Hide_Password_Swift/ViewController.swift
//
//  ViewController.swift
//  Show_Hide_Password_Swift
//
//  Created by mallikarjun on 09/09/19.
//  Copyright © 2019 Mallikarjun H. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var usernameTextField: UILabel!
  
    @IBOutlet weak var passwordTextField: UITextField!
   
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        let hideShow = UIButton(frame: CGRect(x: 0, y: 0, width: 20, height: 20))
        
        hideShow.setImage(UIImage(named: "show"), for: .normal)
        
        passwordTextField.isSecureTextEntry = true
        passwordTextField.rightView = hideShow
        passwordTextField.rightViewMode = .always
        hideShow.addTarget(self, action: #selector(hideShow(_:)), for: .touchUpInside)
        
    }

    
    @objc func hideShow(_ sender: Any?) {
        let hideShow = passwordTextField.rightView as? UIButton
        
        if !passwordTextField.isSecureTextEntry {
            passwordTextField.isSecureTextEntry = true
            hideShow?.setImage(UIImage(named: "show"), for: .normal)
        } else {
            passwordTextField.isSecureTextEntry = false
            hideShow?.setImage(UIImage(named: "hide"), for: .normal)
        }
        passwordTextField.becomeFirstResponder()
    }

    
    @IBAction func loginButtonClicked(_ sender: Any) {

        print("Login Button Clicked")
        //API call
    
    }
    
}

